plumbing/bootstrap.py

```diff
@@ -74,88 +67,79 @@ def clean_error_message(message):
+def bootstrap_pipeline(package, source_folder):
     """Save the pipeline file inside the source folder."""
 
     _, source_name = os.path.split(source_folder)
-    package = load_source_datapackage(source_folder)
     pipeline = {source_name: DEFAULT_PIPELINE}
 
     if 'yaml_error' not in package:
         processors = select_processors(package)
-        save_processor_placeholders(processors)
+        save_processor_placeholders(processors, source_folder)
         save_pipeline(pipeline, source_folder)
 
 
 def save_pipeline(pipeline, source_folder):
-    filepath = os.path.join(source_folder, PIPELINE_SPECS)
+    filepath = os.path.join(source_folder, PIPELINE_FILE)
     with open(filepath, 'w+') as stream:
         stream.write(yaml.dump(pipeline))
-    logging.debug('Created %s', filepath)
+    pipeline_id = os.path.split(source_folder)[-1]
+    logging.info('%s: created pipeline specs', pipeline_id)
 
 
 def select_processors(package):
     """Select the appropriate processors for the pipeline."""
 
-    mode = package['extraction_mode']
+    for resource in package['resources']:
+        mode = resource['extraction_mode']
 
-    if sum(map(mode.values(), int)) > 1:
-        message = '{}: extraction mode is ambiguous'
-        raise ValueError(message.format(package['name']))
+        if sum(map(int, mode.values())) > 1:
+            message = '{}: extraction mode is ambiguous'
+            raise ValueError(message.format(package['name']))
 
-    if mode['download_link']:
-        return ['download_remote_sources']
-    elif mode['web_scraper']:
-        return ['scrape_remote_sources']
-    elif mode['pdf_scraper']:
-        return ['download_pdf_sources', 'scrape_pdf_sources']
-    elif mode['user_interface']:
-        raise NotImplementedError('User interface mode not ready yet.')
-    else:
-        ValueError('No extraction mode specified in the description file.')
+        if any([mode['web_scraper'],
+               mode['pdf_scraper'],
+               mode['user_interface']]):
+            return ['scraper']
+        else:
+            return []
 
 
-def save_processor_placeholders(processors):
+def save_processor_placeholders(processors, source_folder):
     """Drop placeholders for processors inside the source folder."""
 
     for processor in processors:
-        if processor in ('scrape_remote_sources', 'scrape_pdf_sources'):
-            with open(processor + '.py') as script:
-                docstring = "A processor to {}"
-                name = processor.replace('_', ' ')
-                script.write(docstring.format(name))
-                script.write('\n# Help wanted!')
-
-
-def load_source_datapackage(source_folder):
-    """Return the contents of the local datapackage file."""
-
-    datapackage_file = os.path.join(source_folder, DATAPACKAGE_FILE)
-    with open(datapackage_file) as stream:
-        return json.loads(stream.read())
+        if processor in ('scraper', 'scrape_pdf_sources'):
```

Comment:
your empty list in `select_processors` will mean that this function "fails", albeit silently, by doing nothing. would this make it hard to trace a problem?